plugins/modules/docker_container_exec.py

```diff
@@ -0,0 +1,255 @@
+        tty=dict(type='bool', default=False),
+    )
+
+    client = AnsibleDockerClient(
+        argument_spec=argument_spec,
+        min_docker_api_version='1.20',
+        mutually_exclusive=[('argv', 'command')],
+        required_one_of=[('argv', 'command')],
+    )
+
+    container = client.module.params['container']
+    argv = client.module.params['argv']
+    command = client.module.params['command']
+    chdir = client.module.params['chdir']
+    user = client.module.params['user']
+    stdin = client.module.params['stdin']
+    strip_empty_ends = client.module.params['strip_empty_ends']
+    tty = client.module.params['tty']
+
+    if command is not None:
+        argv = shlex.split(command)
+
+    if stdin is not None and client.module.params['stdin_add_newline']:
+        stdin += '\n'
+
+    selectors = None
+    if stdin:
+        selectors = find_selectors(client.module)
+
+    try:
+        exec_data = client.exec_create(
+            container,
+            argv,
+            stdout=True,
+            stderr=True,
+            stdin=bool(stdin),
+            user=user or '',
+            workdir=chdir,
+        )
+        exec_id = exec_data['Id']
+
+        if selectors:
+            exec_socket = client.exec_start(
+                exec_id,
+                tty=tty,
+                detach=False,
+                socket=True,
+            )
+            try:
+                with DockerSocketHandlerModule(exec_socket, client.module, selectors) as exec_socket_handler:
+                    if stdin:
+                        exec_socket_handler.write(to_bytes(stdin))
+
+                    stdout, stderr = exec_socket_handler.consume()
+            finally:
+                exec_socket.close()
+        else:
+            stdout, stderr = client.exec_start(
+                exec_id,
+                tty=tty,
+                detach=False,
+                stream=False,
+                socket=False,
+                demux=True,
+            )
+
+        result = client.exec_inspect(exec_id)
+
+        stdout = to_text(stdout or b'')
+        stderr = to_text(stderr or b'')
+        if strip_empty_ends:
+            stdout = stdout.rstrip('\r\n')
+            stderr = stderr.rstrip('\r\n')
+
+        client.module.exit_json(
+            changed=True,
+            stdout=stdout,
+            stderr=stderr,
+            rc=result.get('ExitCode') or 0,
+        )
+    except NotFound:
+        client.fail('Could not find container "{0}"'.format(container))
+    except APIError as e:
+        if e.response and e.response.status_code == 409:
+            client.fail('The container "{0}" has been paused ({1})'.format(container, e))
+        client.fail('An unexpected docker error occurred: {0}'.format(e), exception=traceback.format_exc())
+    except DockerException as e:
+        client.fail('An unexpected docker error occurred: {0}'.format(e), exception=traceback.format_exc())
```

Comment:
```suggestion
        client.fail('An unexpected docker error occurred: {0}'.format(e), exception=traceback.format_exc())
```
shouldn't we use `to_native()` here and everywhere?

Reply:
Good question. All modules basically use the same error handling (without `to_native`), and so far I never heard about problems with it.

Reply:
I usually prefer to follow the doc when writing new stuff. IMO it's not crucial anyway, so as you like

Reply:
Since this is common to (almost) all modules, I think I'd prefer to fix it in all of them at once.

Reply:
(Also easier to undo for all at once if it turns out to cause problems somewhere...)